src/lib.rs

```diff
@@ -46,7 +46,21 @@ pub mod peripheral;
 pub mod peripherals;
 pub mod prelude;
 pub mod reset;
+
+// mutually exclusive features assert
+#[cfg(all(feature = "rmt-legacy", esp_idf_comp_espressif__onewire_bus_enabled))]
+compile_error!("the onewire component cannot be used with the legacy rmt peripheral");
+
+#[cfg(any(feature = "rmt-legacy", esp_idf_version_major = "4"))]
```

Comment:
This cannot be done so simply. As discussed during the previous iteration, you _do_ need the RMT peripherals which are defined by the `esp_idf_hal::rmt` module. And you need to pass a RMT channel to `OWDriver` in addition to the pin. So you can't just `#ifNdef` the `rmt` module when the `rmt-legacy` feature is not defined. What you need to do instead is to leave the `rmt` module here unconditionally, and then - _inside_ the `rmt` module - protect with an `#ifdef` all of the driver code, but leave the peripheral definitions always defined and valid - even when the legacy RMT driver is `#ifNdef`-ed.

Reply:
Done